test/conformance/tests/sriov_operator.go

```diff
@@ -1039,6 +1039,35 @@ var _ = Describe("[sriov] operator", func() {
 					Expect(err).To(HaveOccurred())
 				})*/
 			})
+			Context("Main PF", func() {
+				It("should work when vfs are used by pods", func() {
+					if !discovery.Enabled() {
+						testNode := sriovInfos.Nodes[0]
+						resourceName := "mainpfresource"
+						sriovDeviceList, err := sriovInfos.FindSriovDevices(testNode)
+						Expect(err).ToNot(HaveOccurred())
+						executorPod := createCustomTestPod(testNode, []string{}, true)
+						mainDeviceForNode := findMainSriovDevice(executorPod, sriovDeviceList)
+						if mainDeviceForNode == nil {
+							Skip("Could not find pf used as gateway")
+						}
+						createSriovPolicy(mainDeviceForNode.Name, testNode, 2, resourceName)
+					}
+
+					mainDevice, resourceName, nodeToTest, ok := discoverResourceForMainSriov(sriovInfos)
```

Comment:
The main reason for this is running discovery mode for the configuration we just applied.
In this way the discovery part is shared between regular and discovery mode, and we use the discovered configuration even in regular mode.

Reply:
the comment was because we are going to create the pods again to find the default GW again but I am fine with this right now if we see the test will take a look of time in not discovery mode we can revisit this